moi.js

```diff
@@ -57,41 +57,55 @@ var moi = new function () {
     this.send = function(action, data) {
         to_send = {}; 
         to_send[action] = data;
-        ws.send(encode(to_send))
+        ws.send(encode(to_send));
     };
-    
+
     /**
      *
      * Verify the browser supports websockets, and if so, initialize the
      * websocket. On construction, this method will send a message over the
      * socket to get all known job information associated with this client.
      *
-     * @param {group_id} A group ID to get initial data from, or null to fetch 
-     * all records associated with the user. 
+     * @param {host} The URL for the websocket, minus the ws:// header, or null
+     * to use the default moi-ws.
+     * @param {group_id} A group ID to get initial data from, or null to fetch
+     * all records associated with the user.
+     * @param {on_open} Optional function for action when websocket is opened.
+     * @param {on_close} Optional function for action when websocket is closed.
+     * @param {on_error} Optional function for action when websocket errors.
      */
-    this.init = function(group_id) {
+    this.init = function(host, group_id, on_open, on_close, on_error) {
```

Comment:
group_id should still be the first argument here, otherwise this is a very non-backwards compatible change.